### PR TITLE
Fixes to T3K test hangs for Mixtral 7x8B

### PR DESCRIPTION
**Pull request by ifabijanic-tt**
### Problem description
Mixtral 8x7B performance-ci-eval-32 test on T3000 was failing on CI with hanging on the prefill phase when tracing is turned on. The issue is shown below:
...
2025-12-17 03:39:30.172 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:313 - Prefill seq len: 128, max_prefill_chunk_size: 4096, trace: True
2025-12-17 03:39:30.291 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:303 - Prefilling User 25 up to 28 tokens
2025-12-17 03:39:30.291 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:313 - Prefill seq len: 128, max_prefill_chunk_size: 4096, trace: True
2025-12-17 03:39:30.409 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:303 - Prefilling User 26 up to 41 tokens
2025-12-17 03:39:30.410 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:313 - Prefill seq len: 128, max_prefill_chunk_size: 4096, trace: True
2025-12-17 03:39:30.528 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:303 - Prefilling User 27 up to 35 tokens
2025-12-17 03:39:30.528 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:313 - Prefill seq len: 128, max_prefill_chunk_size: 4096, trace: True
2025-12-17 03:39:30.647 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:303 - Prefilling User 28 up to 37 tokens
2025-12-17 03:39:30.647 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:313 - Prefill seq len: 128, max_prefill_chunk_size: 4096, trace: True
2025-12-17 03:39:30.765 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:303 - Prefilling User 29 up to 81 tokens
2025-12-17 03:39:30.766 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:313 - Prefill seq len: 128, max_prefill_chunk_size: 4096, trace: True
2025-12-17 03:39:30.884 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:303 - Prefilling User 30 up to 33 tokens
2025-12-17 03:39:30.884 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:313 - Prefill seq len: 128, max_prefill_chunk_size: 4096, trace: True
2025-12-17 03:39:31.002 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:303 - Prefilling User 31 up to 45 tokens
2025-12-17 03:39:31.003 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:313 - Prefill seq len: 128, max_prefill_chunk_size: 4096, trace: True
2025-12-17 03:39:31.121 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:303 - Prefilling User 32 up to 54 tokens
2025-12-17 03:39:31.121 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:313 - Prefill seq len: 128, max_prefill_chunk_size: 4096, trace: True
2025-12-17 03:39:31.240 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:386 - Finished prefill for all users up to 820 tokens, Starting decode...
2025-12-17 03:39:31.240 | INFO     | models.tt_transformers.demo.simple_text_demo:test_demo_text:1009 - Finished prefill warmup
2025-12-17 03:39:31.240 | INFO     | models.tt_transformers.demo.simple_text_demo:test_demo_text:1011 - Starting prefill...
2025-12-17 03:39:31.240 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:303 - Prefilling User 1 up to 393 tokens
2025-12-17 03:39:31.240 | INFO     | models.tt_transformers.tt.generator:prefill_forward_text:313 - Prefill seq len: 1024, max_prefill_chunk_size: 4096, trace: True
context canceled
Error: The action 'Run demo regression tests' has timed out after 50 minutes.

The problem turned out to be memory corruption after tracing. Tracing requires that all memory allocations and ops are known during tracing, but there were some additional memory allocations and ops which happened after the tracing which triggered some nasty memory corruption bugs which were hard to find and which registered only sometimes.

### What's changed
The solution to this problem was to find out if the data formats are correct and to see what is wrong with tracing. When the solution was isolated to the memory allocation problems during tracing then some measures were done. The ops that were previously done outside the forward pass are now done there in order to ensure that all operations are caught during tracing path. There was also some wrong code and comment assumptions which are now also removed. Current solution should fix this memory allocation issues while keeping throughput and TTFT metrics the same as before.

### Checklist

- [x] [![All post-commit tests] https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml (all test fails are devops related and unrelated to this fix)

#### Model tests

If your changes cover model-related code, you should run tests corresponding to affected models and platforms (Single card, T3K, Galaxy). "Choose your pipeline" workflows facilitate running multiple kinds of tests in a single run. Each offers `models-mandatory` and `models-extended` presets.
The former includes a minimal set of tests, to be run always. The latter extends that with additional ones - use your best judgement in deciding which is the most appropriate for your PR.

- [x] (Single card) https://github.com/tenstorrent/tt-metal/actions/runs/20955635467 (one ResNet fail is devops related bug with tt-smi reset failing)
- [x] (T3K) https://github.com/tenstorrent/tt-metal/actions/runs/20936669105 (Mixtral test is now passing without hanging, other tests are unrelated )

### CODE REVIEW

**Review by yieldthought — APPROVED** [review]

**Review by skhorasganiTT — APPROVED** [review]

**Review by gwangTT — APPROVED** [review]

File: models/tt_transformers/tt/generator.py [hunk 1]
```diff
@@ -381,29 +384,11 @@ def prefill_forward_text(
...
-                # For now, assume trace is enabled for embedding models
-                if return_hidden_states:
-                    raise NotImplementedError("return_hidden_states=True requires enable_trace=True")
-                # We have to dispatch copy to host to avoid corruption by the next user's prefill
-                out_list.append(logits.cpu(blocking=False))
-
-        # Process the outputs after all the prefill are done in data parallel mode
+
+            # We have to dispatch copy to host to avoid corruption by the next user's prefill
+            out_list.append(logits.cpu(blocking=False))
```
Comment (skhorasganiTT) [inline models/tt_transformers/tt/generator.py:389]: Was the hidden_states path removed by accident?
